client/my-sites/themes/themes-selection.jsx

```diff
@@ -144,11 +150,15 @@ class ThemesSelection extends Component {
 
 		return (
 			<div className="themes__selection">
-				<QueryThemes query={ query } siteId={ source } />
-				<ThemesSelectionHeader label={ listLabel } count={ themesCount } />
+				{ ! this.props.recommendedThemes && (
+					<>
+						<QueryThemes query={ query } siteId={ source } />
+						<ThemesSelectionHeader label={ listLabel } count={ themesCount } />
+					</>
+				) }
 				<ThemesList
 					upsellUrl={ upsellUrl }
-					themes={ this.props.themes }
+					themes={ this.props.recommendedThemes || this.props.themes }
```

Comment:
Why do we need to pass both  `recommendedThemes` and `themes` into the component? Is there a way to abstract it so that the component only takes a `themes` prop, and then `recommendedThemes` are given to that prop in the right context? (I may also be confused about what this does :)

Reply:
Here we are passing one of the two into the `ThemesList` component as the `themes` prop.  So if we passed `recommendedThemes` into `ThemesSelection` like we do from the `RecommendedThemes` component, then `ThemesList` receives that as the `themes` prop.  If there is no `recommendedThemes` prop (how the component is normally used), then it falls back to pass `props.themes` connection from state.  So in the case of `ThemesList` it is always receiving only one of those as the `themes` prop.   

Or were you saying that the the `ThemesSelection` component itself should be cleaned up in a way that we aren't passing the recommended list down?  I suppose we could make this a condition in the `connect` HOC to determine what `themes` should be so we don't need to have `recommendedThemes` inside the actual component class.  But we do find `recommendedThemes` helpful as a boolean check in the component class to condition things like `incrementPage` function and whether or not we should render certain other components.  So Im not sure if thats going to be very helpful either. 🤔